packages/web3-eth-ens/src/registry.ts

```diff
@@ -0,0 +1,183 @@
+import { inputAddressFormatter, ReceiptInfo } from 'web3-common';
+import { Contract, NonPayableCallOptions } from 'web3-eth-contract';
+import { Address, isHexStrict, sha3Raw } from 'web3-utils';
+import REGISTRY from './ABI/registry';
+import { RESOLVER } from './ABI/resolver';
+import { registryAddress } from './config';
+import { namehash } from './utils';
+
+export class Registry {
+	private readonly contract: Contract<typeof REGISTRY>;
+
+	public constructor(customRegistryAddress?: Address) {
+		// TODO for contract, when eth.net is finished we can check network
+		this.contract = customRegistryAddress
+			? new Contract(REGISTRY, customRegistryAddress)
+			: new Contract(REGISTRY, registryAddress);
+	}
+	public async getOwner(name: string) {
+		try {
+			const result = await this.contract.methods.owner(namehash(name)).call();
+
+			return result;
+		} catch (error) {
+			throw new Error(); // TODO: TransactionRevertError Needs to be added after web3-eth call method is implemented
+		}
+	}
+
+	public async setOwner(
+		name: string,
+		address: Address,
+		txConfig: NonPayableCallOptions, // TODO: web3-eth txconfig should be replaced with sendTransaction type
+	) {
+		try {
+			const receipt = await this.contract.methods
```

Comment:
this should return [`PromiEvent<TransactionReceipt | TransactionRevertInstructionError>`](https://web3js.readthedocs.io/en/v1.7.0/callbacks-promises-events.html#promievent) and same for all set functions , as in [1.x](https://web3js.readthedocs.io/en/v1.7.0/web3-eth-contract.html#id36) , 

I think we should not await here and directly return promise, and same for all set functions ,